### PR TITLE
Adapt tests for IPv6 hosts

### PR DESCRIPTION
**Pull request by benjamreis**
- add brackets around IP in URI when necessary
- ignore tests not suporting IPv6

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -176,6 +177,17 @@ def xfail_on_xcpng_8_3(host, request):
     if host.xcp_version >= version.parse("8.3"):
         request.node.add_marker(pytest.mark.xfail)
 
+@pytest.fixture(scope='session')
+def host_no_ipv6(host):
+    if is_ipv6(host.hostname_or_ip):
+        pytest.skip(f"This test requires an IPv4 XCP-ng")
+
+@pytest.fixture(scope='function')
+def xfail_on_xcpng_ipv6(host, request):
```
Comment (stormi) [inline conftest.py:186]: We don't need this fixture anymore, right?

File: tests/storage/moosefs/test_moosefs_sr.py [hunk 2]
```diff
@@ -31,6 +31,7 @@ def test_create_moosefs_sr_without_mfsmount(self, host, moosefs_device_config):
             sr.destroy()
             assert False, "MooseFS SR creation should failed!"
 
+    @pytest.mark.usefixtures("host_no_ipv6")
```
Comment (stormi) [inline tests/storage/moosefs/test_moosefs_sr.py:34]: Everytime we use this fixture, I think there should be a comment explaining why the test has to be skipped in IPv6 mode.

File: conftest.py (hunk 1 above)
Comment (benjamreis) [reply conftest.py:186]: Right, I thought about remove and then thought it might be handy in the future.
I can remove it if you'd like.

**Comment by stormi** [comment]
LGTM :+1:

**Review by stormi — APPROVED** [review]

**Pull request merged**